ledger/artifactmanager/artifactmanager.go

```diff
@@ -40,10 +40,17 @@ type LedgerArtifactManager struct {
 	db                         *storage.DB
 	DefaultBus                 core.MessageBus                 `inject:""`
 	PlatformCryptographyScheme core.PlatformCryptographyScheme `inject:""`
+	codeCacheLock              *sync.Mutex
+	codeCache                  map[core.RecordRef]*cacheEntry
 
 	getChildrenChunkSize int
 }
 
+type cacheEntry struct {
+	sync.Mutex
```

Comment:
After a brief description we agreed that this approach is OK since this lock prevents sending multiple requests. However a corresponding comment should be added here.